### ISSUE TITLE
`unicorn/prefer-spread` treats `string.slice(0)` like an array

### ISSUE THREAD

**Issue by yukulele** [sindresorhus/eslint-plugin-unicorn#2231]
`unicorn/prefer-spread` give wrong fix of `string.slice(0)`

rule: `unicorn/prefer-spread`

code:
```js
const original = 'foo'
const copy = original.slice(0)
```

actual fix:
```js
const original = 'foo'
// Prefer the spread operator over `Array#slice()`
const copy = [...original]
```
expected fix:
```js
const original = 'foo'
// Primitives are immutables
const copy = original
```

**Comment by fisker** [comment]
>  give wrong fix of string.splice(0)

There is no `String#splice`

> const copy = original.slice(0)

Don't understand what purpose of slicing a string, shoudn't we use `const copy = original`?

**Comment by fregante** [comment]
- Merged into https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2041

The issue is that most (all?) unicorn rules don't/can't use types to avoid false positives.
